src/LineaStateBridge.sol

```diff
@@ -1,17 +1,251 @@
+    /// to the WorldID Identity Manager contract away
+    /// @param previousOwner The previous owner of the LineaWorldID contract
+    /// @param messageService The new owner of the LineaWorldID contract
+    /// @param remoteAddress Whether the ownership transfer is local (Linea/Linea Stack chain EOA/contract)
+    /// or an Ethereum EOA or contract
+    event OwnershipTransferredLinea(
+        address indexed previousOwner, address indexed messageService, address remoteAddress
+    );
+
+    /// @notice Emitted when the StateBridge sends a root to the LineaWorldID contract
+    /// @param root The root sent to the LineaWorldID contract on the Linea Stack chain
+    event RootPropagated(uint256 root);
+
+    /// @notice Emitted when the StateBridge sets the root history expiry for OpWorldID, LineaWorldID and PolygonWorldID
+    /// @param rootHistoryExpiry The new root history expiry
+    event SetRootHistoryExpiry(uint256 rootHistoryExpiry);
+
+    /// @notice Emitted when the StateBridge sets the gas limit for sendRootOp
+    /// @param _lineaGasLimit The new _lineaGasLimit  for sendRootOp
+    event SetGasLimitPropagateRoot(uint32 _lineaGasLimit);
+
+    /// @notice Emitted when the StateBridge sets the gas limit for SetRootHistoryExpiry
+    /// @param _lineaGasLimit The new _lineaGasLimit  for SetRootHistoryExpiry
+    event SetGasLimitSetRootHistoryExpiry(uint32 _lineaGasLimit);
+
+    /// @notice Emitted when the StateBridge sets the gas limit for transferOwnershipOp
+    /// @param _lineaGasLimit The new _lineaGasLimit  for transferOwnershipOptimism
+    event SetGasLimitTransferOwnershipLinea(uint32 _lineaGasLimit);
+
+    ///////////////////////////////////////////////////////////////////
+    ///                            ERRORS                           ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice Emitted when an attempt is made to renounce ownership.
+    error CannotRenounceOwnership();
+
+    /// @notice Emitted when an attempt is made to set the gas limit to zero
+    error GasLimitZero();
+
+    /// @notice Emitted when an attempt is made to set an address to zero
+    error AddressZero();
+
+    ///////////////////////////////////////////////////////////////////
+    ///                         CONSTRUCTOR                         ///
+    ///////////////////////////////////////////////////////////////////
+
+    /// @notice constructor
+    /// @param _worldIDIdentityManager Deployment address of the WorldID Identity Manager contract
+    /// @param _lineaWorldIDAddress Address of the Linea contract that will receive the new root and timestamp
+    /// @param _crossDomainMessenger L1CrossDomainMessenger contract used to communicate with the desired Linea
+    /// Stack network
+    /// @custom:revert if any of the constructor params addresses are zero
+    constructor(address _worldIDIdentityManager, address _lineaWorldIDAddress, address _crossDomainMessenger) {
```

Comment:
> _crossDomainMessenger
I think if we use Linea terminology, we may name it "messageService", correct me if I'm wrong

Reply:
alright, I think this would work, will make changes to it now to `messageService`